pymongo/collection.py

```diff
@@ -2280,6 +2279,178 @@ def index_information(
+          set.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+        """
+        if name is None:
+            pipeline: _Pipeline = [{"$listSearchIndexes": {}}]
+        else:
+            pipeline = [{"$listSearchIndexes": {"name": name}}]
+        return self.aggregate(pipeline, session, comment=comment, **kwargs)
+
+    def create_search_index(
+        self,
+        description: SearchIndexModel,
+        session: Optional["ClientSession"] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> str:
+        """Create a single search index for the collection.
+
+        :Parameters:
+          - `description` - The index description for the new search index.
+          - `session` (optional): a
+            :class:`~pymongo.client_session.ClientSession`.
+          - `comment` (optional): A user-provided comment to attach to this
+            command.
+          - `**kwargs` (optional): optional arguments to the createSearchIndexes
+            command (like maxTimeMS) can be passed as keyword arguments.
+
+        :Returns:
+          The name of the new search index.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+        """
+        return self.create_search_indexes([description], session, comment, **kwargs)[0]
+
+    def create_search_indexes(
+        self,
+        descriptions: List[SearchIndexModel],
+        session: Optional["ClientSession"] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Create multiple search indexes for the current collection.
+
+        :Parameters:
+          - `descriptions` - A list of `SearchIndexDescription`s for the new search indexes.
+          - `session` (optional): a
+            :class:`~pymongo.client_session.ClientSession`.
+          - `comment` (optional): A user-provided comment to attach to this
+            command.
+          - `**kwargs` (optional): optional arguments to the createSearchIndexes
+            command (like maxTimeMS) can be passed as keyword arguments.
+
+        :Returns:
+            A list of the newly created search index names.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+        """
+        if comment is not None:
+            kwargs["comment"] = comment
+        names = []
+
+        def gen_indexes():
+            for index in descriptions:
+                if not isinstance(index, SearchIndexModel):
+                    raise TypeError(
+                        f"{index!r} is not an instance of pymongo.operations.SearchIndexModel"
+                    )
+                doc = index.document
+                if "name" in doc:
+                    names.append(doc["name"])
+                yield doc
+
+        cmd = SON([("createSearchIndexes", self.name), ("indexes", list(gen_indexes()))])
+        cmd.update(kwargs)
+
+        self.database.command(
```

Comment:
I suspect this needs to use _socket_for_writes as well.

Reply:
Fixed